src/org/geekhub/json/JsonSerializer.java

```diff
@@ -70,7 +77,21 @@ public static Object serialize(Object o) {
      * @throws InstantiationException
      */
     private static JSONObject toJsonObject(Object o) throws Exception {
-        //implement me
-        return null;
+        Class<? extends Object> catClass = o.getClass();
+        Field[] fields = catClass.getDeclaredFields();
+        JSONObject jsonObject = new JSONObject();
+        for(Field field : fields){
+            field.setAccessible(true);
+            if(simpleTypes.contains(field.getType())){
```

Comment:
You don't need to check simpleTypes.contains... You already have this check in serialize method.